src/PublicShareSidebarTrigger.vue

```diff
@@ -4,62 +4,42 @@
 -->
 
 <template>
-	<div class="button-wrapper">
-		<NcButton type="tertiary-on-primary"
-			:title="ariaLabel"
-			:aria-label="ariaLabel"
-			@click="$emit('click')">
-			<template #icon>
-				<MessageText fill-color="var(--color-background-plain-text)" :size="20" />
-			</template>
-		</NcButton>
-	</div>
+	<NcHeaderButton id="talk-sidebar-trigger"
+		:title="ariaLabel"
+		:aria-label="ariaLabel"
+		@click="emit('click')">
+		<template #icon>
+			<MessageText :size="20" />
+		</template>
+	</NcHeaderButton>
 </template>
 
-<script>
+<script setup lang="ts">
+import { computed, UnwrapNestedRefs } from 'vue'
+
 import MessageText from 'vue-material-design-icons/MessageText.vue'
 
 import { t } from '@nextcloud/l10n'
 
-import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
-
-export default {
-	name: 'PublicShareSidebarTrigger',
-
-	components: {
-		MessageText,
-		NcButton,
-	},
+import NcHeaderButton from '@nextcloud/vue/dist/Components/NcHeaderButton.js'
 
-	props: {
-		sidebarState: {
-			type: Object,
-			required: true,
-		},
-	},
+const props = defineProps<{
+	sidebarState: UnwrapNestedRefs<{ isOpen: boolean }>
+}>()
```

Comment:
Why are you explicitly unrawpping this prop?

Reply:
That's just the same type as declared for reactive()
![image](https://github.com/user-attachments/assets/33581b16-cb27-425e-99a9-de37df0665be)